Framework/Algorithms/src/XDataConverter.cpp

```diff
@@ -103,7 +103,7 @@ void XDataConverter::setXData(API::MatrixWorkspace_sptr outputWS,
         }
       }
     }
-    outputWS->setX(index, m_cachedX);
+    outputWS->setSharedX(index, m_cachedX);
   } else {
     const MantidVec &xBoundaries = inputWS->readX(index);
     MantidVec &xPoints = outputWS->dataX(index);
```

Comment:
I have actually asked about those before. The problem is that some of the functions involved (such as `Kernel::VectorHelper::convertToBinCentre` called from `ConvertToPointData::calculateXPoints` requires parameters in MantidVec form. I was not able to use the new functions since they were not being given in that form. Changing `Kernel::VectorHelper::convertToBinCentre` parameters is impractical as it is used by many other algorithms.